src/selectors/entities/groups.test.js

```diff
@@ -134,7 +137,12 @@ describe('Selectors.Groups', () => {
 
     it('getGroupsNotAssociatedToChannel', () => {
         const expected = Object.entries(testState.entities.groups.groups).filter(([groupID]) => !channelAssociatedGroupIDs.includes(groupID)).map(([, group]) => group);
-        assert.deepEqual(Selectors.getGroupsNotAssociatedToChannel(testState, channelID), expected);
+        assert.deepEqual(Selectors.getGroupsNotAssociatedToChannel(testState, channelID, teamID), expected);
+
+        const cloneState = JSON.parse(JSON.stringify(testState));
+        cloneState.entities.teams.teams[teamID].group_constrained = true;
+        cloneState.entities.teams.groupsAssociatedToTeam[teamID].ids = [expectedAssociatedGroupID1];
```

Comment:
can we just use destructuring to clone the new state?

Reply:
> can we just use destructuring to clone the new state?

I tried that first but our `deepFreezeAndThrowOnMutation` function rejected it. Let me try again because it *should* work.

Reply:
Looking into it a bit more it looks like the custom `__defineSetter__` methods are cloned across when doing `const cloneState = {...testState}`, which explains why it doesn't work. To clone the object like that I think we would have to re-define the methods, something like this:

```js
cloneState.entities.teams.teams[teamID].__defineSetter__("group_constrained", someFunc);
cloneState.entities.teams.groupsAssociatedToTeam[teamID].__defineSetter__("ids", someFunc);
```

Doesn't seem worth it.

Reply:
I meant something like this @mkraft - up to you either way im 0/5 on it 
```
const modifiedState = {
            ...testState,
            entities: {
                ...testState.entities,
                teams: {
                    ...testState.entities.teams,
                    teams: {
                        ...testState.entities.teams.teams,
                        [teamID]: {
                            ...testState.entities.teams.teams[teamID],
                            group_constrained: true,
                        },
                    },
                    groupsAssociatedToTeam: {
                    ...testState.entities.teams.groupsAssociatedToTeam,
                        [teamID]: {
                            ...testState.entities.teams.groupsAssociatedToTeam[teamID],
                            ids: [expectedAssociatedGroupID1],
                        },
                    },
                },
            },
        };
```